zio-http/src/main/scala/zhttp/http/Cookie.scala

```diff
@@ -142,58 +153,117 @@ object Cookie {
-    val (name, content)   = splitNameContent(first)
-
-    val cookie =
-      if (name.trim == "" && content.isEmpty) Option.empty[Cookie]
-      else Some(Cookie(name, content.getOrElse("")))
-
-    other.map(splitNameContent).map(t => (t._1.toLowerCase, t._2)).foldLeft(cookie) {
-      case (Some(c), ("expires", Some(v)))  => parseDate(v).map(c.withExpiry(_))
-      case (Some(c), ("max-age", Some(v)))  => Try(v.toLong).toOption.map(c.withMaxAge(_))
-      case (Some(c), ("domain", v))         => Some(c.withDomain(v.getOrElse("")))
-      case (Some(c), ("path", v))           => Some(c.withPath(Path(v.getOrElse(""))))
-      case (Some(c), ("secure", _))         => Some(c.withSecure)
-      case (Some(c), ("httponly", _))       => Some(c.withHttpOnly)
-      case (Some(c), ("samesite", Some(v))) =>
-        v.trim.toLowerCase match {
-          case "lax"    => Some(c.withSameSite(SameSite.Lax))
-          case "strict" => Some(c.withSameSite(SameSite.Strict))
-          case "none"   => Some(c.withSameSite(SameSite.None))
-          case _        => Some(c)
+  def decodeResponseCookie(headerValue: String): Option[Cookie] =
+    Option(unsafeDecodeResponseCookie(headerValue))
+
+  private[http] def unsafeDecodeResponseCookie(headerValue: String): Cookie = {
+    var name: String              = null
+    var content: String           = null
+    var expires: Instant          = null
+    var maxAge: Option[Long]      = None
+    var domain: String            = null
+    var path: Path                = null
+    var secure: Boolean           = false
+    var httpOnly: Boolean         = false
+    var sameSite: Cookie.SameSite = null
+
+    val headerLength = headerValue.length
+
+    // iterate over all cookie fields (until next semicolon)
+    var curr = 0
+    var next = 0
+
+    while (next >= 0 && curr < headerLength) {
+      next = headerValue.indexOf(';', curr)
+      if (next < 0) {
+        next = headerLength
+      }
+
+      // skip whitespaces one by one to avoid trim allocations
+      if (headerValue.charAt(curr) == ' ') {
+        curr = curr + 1
+      } else {
+        // decode name and content first
+        if (name == null) {
+          val splitIndex = headerValue.indexOf('=', curr)
+          if (splitIndex >= 0 && splitIndex < next) {
+            name = headerValue.substring(0, splitIndex)
+            content = headerValue.substring(splitIndex + 1, next)
+          } else {
+            name = headerValue.substring(0, next)
+          }
+        } else if (headerValue.regionMatches(true, curr, fieldExpires, 0, fieldExpires.length)) {
+          expires =
+            try { Instant.parse(headerValue.substring(curr + 8, next)) }
+            catch { case _: Throwable => null }
+        } else if (headerValue.regionMatches(true, curr, fieldMaxAge, 0, fieldMaxAge.length)) {
+          maxAge = Try(headerValue.substring(curr + 8, next).toLong).toOption
```

Comment:
We can create a global `try` `catch` block for this method.

Reply:
@tusharmath , a couple of questions here:
1) would it be enough to wrap with a try each attempt to decode a field?

2) What should we do with the exception, discard it completely?

Reply:
1. We should have only 1 top level try catch block inside the `safe` implementation.
2. Incase of failure we should throw an exception in the `unsafe` implementation. The safe implementation could do either of the two things —
   a. Ignore the error and return a `None`.
   b. Capture the error and return a `Left[Throwable]`

Reply:
Ok, changed to catch the exception in decodeResponseCookie but keeping the Option[Cookie] signature